File: bootcamp_c_cpp_dsa-main/bootcamp_c_cpp_dsa-main/CPP/codes/p36.cpp
#include<iostream>
class Demo{
	public:
		Demo(){
			std::cout<< "No-args Constructor" <<std::endl;
		}
		Demo(int x){
			std::cout<< "Para Constructor" <<std::endl;
		}
		Demo(Demo &xyz){
			std::cout<< "Copy Constructor" <<std::endl;
		}
};
int main(){
	Demo obj1;
	Demo obj3(10);
	Demo obj2(obj1);
	Demo obj4=obj1;
}

File: java-main/java-main/Collection/p36.java
class Demo{
	String str;
	Demo(String str){
		this.str=str;
	}
	public String toString(){
		return str;
	}
	public void finalise(){
		System.out.println("Notify");
	}
}
class GCDemo{
	public static void main(String[]args){
		Demo obj1=new Demo("C2W");
		Demo obj2=new Demo("Bincaps");
		Demo obj3=new Demo("Incubetors");
	
		
		System.out.println(obj1);
		System.out.println(obj2);
		System.out.println(obj3);

		obj1=null;
//		obj2=null;

		System.gc();

		System.out.println("In main");
	}
}

		

